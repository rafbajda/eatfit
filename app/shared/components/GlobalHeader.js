import React from 'react';
import { Header, Left, Button, Icon, Body, Right, Thumbnail } from 'native-base';
import { Image } from 'react-native-elements';
import Constants from 'expo-constants';
import { globalGreen } from '../constants/colors';
import screens from '../../navigation/screens';

const logoImage = require('../../assets/images/eatfit_without_icon.png');

const GlobalHeader = props => {
    const { nav, avatar } = { ...props };
    return (
        <Header
            style={{
                backgroundColor: globalGreen,
                marginTop: Constants.statusBarHeight
            }}
        >
            <Left style={{ flex: 1 }}>
                <Button transparent onPress={() => nav.openDrawer()}>
                    <Icon name="menu" />
                </Button>
            </Left>
            <Body style={{ flex: 1 }}>
                <Image
                    source={logoImage}
                    style={{ width: 120, height: 25 }}
                    placeholderStyle={{ backgroundColor: globalGreen }}
                />
            </Body>
            <Right style={{ flex: 1 }}>
                <Button onPress={() => nav.navigate(screens.Profile)} transparent>
                    <Thumbnail
                        small
                        source={{
                            uri: avatar
                        }}
                    />
                </Button>
            </Right>
        </Header>
    );
};

export default GlobalHeader;
