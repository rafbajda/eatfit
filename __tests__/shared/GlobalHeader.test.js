import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import GlobalHeader from '../../app/shared/components/GlobalHeader';

describe('GlobalHeader snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<GlobalHeader />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
