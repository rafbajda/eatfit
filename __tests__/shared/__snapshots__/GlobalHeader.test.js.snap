// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalHeader snapshot renders component 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        },
        Object {
          "backgroundColor": "#1db954",
          "marginTop": 44,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginLeft": -3,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 4,
            "paddingRight": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "position": "relative",
          }
        }
      >
        <Image
          onLoad={[Function]}
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 25,
                "width": 120,
              },
            ]
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        />
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1db954",
                "height": 25,
                "justifyContent": "center",
                "width": 120,
              }
            }
            testID="RNE__Image__placeholder"
          />
        </View>
        <View
          style={
            Object {
              "height": 25,
              "width": 120,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 50,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginRight": -9,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Image
          small={true}
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
